File: Hashmap & Set/Unique Number of Occurences/Unique.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        checker = set()
        hashmap = defaultdict(int)

        for num in arr:
            hashmap[num] += 1

        for num in hashmap.values():
            if num in checker:
                return False
            checker.add(num)

        return True

File: Hashmap & Set/Unique Number of Occurences/Unique.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Set<Integer> set = new HashSet<>();
        Map<Integer,Integer> hashmap = new HashMap<>();

        for (int num: arr){
            hashmap.put(num, hashmap.getOrDefault(num,0) + 1);
        }

        for (int num: hashmap.values()){
            if (set.contains(num)){
                return false;
            }
            set.add(num);
        }
        return true;
    }
}